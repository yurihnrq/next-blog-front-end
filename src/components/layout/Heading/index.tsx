import { clsx } from 'clsx';
import { ElementType, HTMLAttributes } from 'react';

export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
  size?: 'small' | 'medium' | 'large';
}

/**
 * Renders a heading with the given size.
 *
 * size: 'small' (h3) | 'medium' (h2) | 'large' (h1)
 */
export function Heading({
  className,
  children,
  size = 'large',
  ...rest
}: HeadingProps) {
  let Comp: ElementType = 'h1';

  if (size === 'medium') {
    Comp = 'h2';
  }

  if (size === 'small') {
    Comp = 'h3';
  }

  return (
    <Comp
      className={clsx(
        className,
        'font-bold',
        {
          'text-xl': size === 'small',
          'text-2xl': size === 'medium',
          'text-3xl': size === 'large'
        },
        'dark:text-slate-100'
      )}
      {...rest}
    >
      {children}
    </Comp>
  );
}
