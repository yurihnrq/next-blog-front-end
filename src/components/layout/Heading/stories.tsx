import { Meta, StoryObj } from '@storybook/react';
import { Heading, HeadingProps } from '.';

const meta: Meta<HeadingProps> = {
  title: 'Components/Layout/Heading',
  component: Heading
};

export default meta;

export const Default: StoryObj<HeadingProps> = {
  args: {
    size: 'medium',
    children: 'Heading'
  }
};
