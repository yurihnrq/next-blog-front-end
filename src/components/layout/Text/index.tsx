import { Slot } from '@radix-ui/react-slot';
import clsx from 'clsx';
import { HTMLAttributes } from 'react';

export interface TextProps extends HTMLAttributes<HTMLElement> {
  size?: 'small' | 'medium' | 'large';
  asChild?: boolean;
}

export function Text({
  size,
  asChild,
  children,
  className,
  ...rest
}: TextProps) {
  const Comp = asChild ? Slot : 'p';

  return (
    <Comp
      className={clsx(className, {
        'text-sm': size === 'small',
        'text-base': size === 'medium',
        'text-lg': size === 'large'
      })}
      {...rest}
    >
      {children}
    </Comp>
  );
}
