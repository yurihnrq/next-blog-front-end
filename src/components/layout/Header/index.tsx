import { NewspaperIcon } from '@heroicons/react/24/solid';
import Link from 'next/link';

import { Heading } from '../Heading';

export function Header() {
  return (
    <header className='flex flex-row w-full py-2 px-4 justify-between'>
      <Link
        href='/'
        className='flex flex-row gap-2 justify-center items-center'
      >
        <NewspaperIcon className='text-purple-600 h-12' />
        <Heading size='large'>Next Blog</Heading>
      </Link>
      <nav className='flex flex-row text-medium items-center justify-end gap-4'>
        <Link href='/'>Home</Link>
        <Link href='/latest'>Ãšltimos posts</Link>
        <Link href='/most-read'>Mais lidos</Link>
      </nav>
    </header>
  );
}
