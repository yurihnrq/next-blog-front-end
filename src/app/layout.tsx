import { ReactNode } from 'react';
import { Roboto } from 'next/font/google';

import { Header } from '@src/components/layout/Header';

import './globals.css';

const roboto = Roboto({
  weight: '400',
  subsets: ['latin-ext']
});

export const metadata = {
  title: 'Next Blog',
  description: 'Blog built with Next.js'
};

export interface RootLayoutProps {
  children: ReactNode;
}

/**
 * Base layout for all pages.
 */
export default function RootLayout({ children }: RootLayoutProps) {
  return (
    <html lang='pt-BR' className={roboto.className}>
      <body>
        <Header />
        <main className='max-w-7xl mx-auto mt-4'>{children}</main>
      </body>
    </html>
  );
}
