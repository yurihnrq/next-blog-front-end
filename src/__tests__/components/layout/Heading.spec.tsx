import { render } from '@testing-library/react';

import { Heading } from '@src/components/layout/Heading';

describe('Heading', () => {
  it('should render correctly', () => {
    const { container } = render(<Heading>Heading</Heading>);

    expect(container).toMatchSnapshot();
  });

  it('should render as h1 when using default size', () => {
    const { container } = render(<Heading>Heading</Heading>);

    expect(container.querySelector('h1')).toBeInTheDocument();
  });

  it('should render as h1 when size is large', () => {
    const { container } = render(<Heading size='large'>Heading</Heading>);

    expect(container.querySelector('h1')).toBeInTheDocument();
  });

  it('should render as h2 when size is medium', () => {
    const { container } = render(<Heading size='medium'>Heading</Heading>);

    expect(container.querySelector('h2')).toBeInTheDocument();
  });

  it('should render as h3 when size is small', () => {
    const { container } = render(<Heading size='small'>Heading</Heading>);

    expect(container.querySelector('h3')).toBeInTheDocument();
  });
});
