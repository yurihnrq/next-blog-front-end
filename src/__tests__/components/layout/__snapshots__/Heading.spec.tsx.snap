// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render correctly 1`] = `
<div>
  <h1
    class="font-bold text-3xl dark:text-slate-100"
  >
    Heading
  </h1>
</div>
`;
